### PR TITLE
halstack-browser repo is up to date OC-12874

### PR DESCRIPTION
**Pull request by kukoc**
https://jira.dxc.com/browse/OC-12874

- Refactor of the library replacing @diaas/api-sdk by halstack-client
- Github actions:
        - Publish next version  
        - Publish release 
        - build and test on PR
        - HalUI COmponent test

### CODE REVIEW

File: .github/workflows/publish-next.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Publish Next Version
+
+on:
+  push:
+    branches:
+      - aef-actions
```
Comment (marcialfps) [inline .github/workflows/publish-next.yml]: Change to `master`.

File: example/src/App.js [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
-import HalUI from "@diaas/hal-ui";
+import HalUI from "@dxc-technology/halstack-browser";
+//import HalUI from "./hal/hal-ui/HalUI";
 import "./App.css";
 
-const API_PROFILE_ID = ""; // From DXC Developer Central
-const API_USER_NAME = ""; // From DXC Developer Central
-const YOUR_API_KEY = ""; // From DXC Developer Central
+
+const YOUR_API_KEY = 'RlbIOdmhnj9Pd85G8fwFA5fdJWYcXtHq7my0wNa1'; // From DXC Developer Central
```
Comment (marcialfps) [inline example/src/App.js]: I think the API key is private, you should remove it.

File: lib/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+## Halstack Browser in Action
+
+See [this demo app using Halstack browser](https://hal.us.insurance.dxc.com/). Paste your hypermedia-based API endpoint and start browsing!
+
+## Use
+### Installation
+
+```sh
+npm install --save TBD
```
Comment (marcialfps) [inline lib/README.md]: Change `TBD`.

File: lib/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+### Properties
+| Name      | Type          | Default | Description                                                                 |
+| :-------- | :------------ | :------ | :-------------------------------------------------------------------------- |
+| `url`     | `String`      | None    | Required. This property contains the base URL of your HAL API.              |
+| `headers` | `JSON Object` | None    | Contains an object with the headers to be sent to every request to the API. |
+
+## Develop
+In the first place, clone this repository.
+```bash
+git clone https://github.dxc.com/DIaaS/diaas-hal-ui.git
```
Comment (marcialfps) [inline lib/README.md]: Change repository url.

File: lib/src/hal-ui/HalUI.jsx [hunk 5]
```diff
@@ -28,14 +28,17 @@ export default class HalUI extends Component {
   setBaseResource(url) {
     this.setState(() => ({ halResource: undefined, baseURL: url, isLoading: true }));
     const { headers } = this.props;
-    axios({
+    /*axios({
```
Comment (marcialfps) [inline lib/src/hal-ui/HalUI.jsx]: Remove unused code.

File: lib/src/hal-ui/Operation.jsx [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import styled from "styled-components";
-import axios from "axios";
-import { halResource } from "@diaas/api-sdk";
+//import axios from "axios";
```
Comment (marcialfps) [inline lib/src/hal-ui/Operation.jsx]: Remove import.

File: lib/src/hal-ui/Operation.jsx [hunk 7]
```diff
@@ -22,7 +22,7 @@ export default class Operation extends React.Component {
     const { method, url, headers } = this.props;
     const { body } = this.state;
     this.setState(() => ({ response: null, isLoading: true }));
-    axios({
+    /*axios({
```
Comment (marcialfps) [inline lib/src/hal-ui/Operation.jsx]: Remove unused code.

File: lib/testingActions.md [hunk 8]
```diff
@@ -0,0 +1 @@
+X
```
Comment (marcialfps) [inline lib/testingActions.md]: This is a file for testing purposes, so you need to remove it before merging.

**Review by marcialfps — CHANGES_REQUESTED** [review]
- Why do you have a `yarn.lock` file in the example directory?

**Comment by kukoc** [comment]
> * Why do you have a `yarn.lock` file in the example directory?

I think we should keep the lock files so the security bots can do a better job scanning vulnerabilities. 

But maybe, you're right,  we don't really care about the example part

File: example/src/App.js [hunk 9]
```diff
@@ -1,30 +1,26 @@
 import React, { Component } from "react";
-import HalUI from "@diaas/hal-ui";
+import HalUI from "@dxc-technology/halstack-browser";
+//import HalUI from "./hal/hal-ui/HalUI";
```
Comment (marcialfps) [inline example/src/App.js]: Remove this comment.

File: lib/README.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+## Halstack Browser in Action
+
+See [this demo app using Halstack browser](https://hal.us.insurance.dxc.com/). Paste your hypermedia-based API endpoint and start browsing!
+
+## Use
+### Installation
+
+```sh
+npm install @diaas/hal-browser --save
```
Comment (marcialfps) [inline lib/README.md]: Change to `@dxc-technology/halstack-browser`

File: lib/README.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
...
+```
+### Implementation example
+
+```js
+// import the HalUI component
+import HalUI from "@diaas/hal-browser";
```
Comment (marcialfps) [inline lib/README.md]: Change to `@dxc-technology/halstack-browser`

**Review by marcialfps — CHANGES_REQUESTED** [review]

**Review by marcialfps — APPROVED** [review]
💯

**Pull request merged**